test/Interop/Cxx/operators/member-inline-silgen.swift

```diff
@@ -32,3 +32,54 @@ public func call(_ wrapper: inout AddressOnlyIntWrapper) -> Int32 { wrapper() }
 // CHECK: end_access [[SELFACCESS]] : $*AddressOnlyIntWrapper
 
 // CHECK: sil [clang AddressOnlyIntWrapper.callAsFunction] [[NAME]] : $@convention(c) (@inout AddressOnlyIntWrapper) -> Int32
+
+public func index(_ arr: inout ReadOnlyIntArray, _ arg: Int32) -> Int32 { arr[arg] }
+
+// CHECK: sil @[[FUNCNAME:.*]] : $@convention(thin) (@inout ReadOnlyIntArray, Int32) -> Int32 {
+// CHECK: bb0([[SELF:%.*]] : $*ReadOnlyIntArray, [[INDEX:%.*]] : $Int32):
+// CHECK:   [[SELFACCESS:%.*]] = begin_access [modify] [static] [[SELF]] : $*ReadOnlyIntArray
+// CHECK:   [[SELFACCESS2:%.*]] = begin_access [modify] [static] [[SELFACCESS]] : $*ReadOnlyIntArray
+// CHECK:   [[OP:%.*]] = function_ref [[READCLASSNAME:@(_ZNK16ReadOnlyIntArrayixEi|\?\?AReadOnlyIntArray@@QEBAAEBHH@Z)]] : $@convention(c) (@inout ReadOnlyIntArray, Int32) -> UnsafePointer<Int32>
+// CHECK:   [[PTR:%.*]] = apply [[OP]]([[SELFACCESS2]], [[INDEX]]) : $@convention(c) (@inout ReadOnlyIntArray, Int32) -> UnsafePointer<Int32>
+// CHECK:   end_access [[SELFACCESS2]] : $*ReadOnlyIntArray
```

Comment:
The checks from L44 to L47 aren't really helpful, imho. We're not testing that the compiler knows how to properly emit load instructions here, we're testing that it knows how to call the right C++ function. I think you can get rid of them. (The only reason to keep the begin_access instructions are so that we know the C++ function is called with the correct arguments.)

Reply:
Fair point, removed these checks.